src/compute-types/src/dataflows.rs

```diff
@@ -67,19 +68,26 @@ pub struct DataflowDescription<P, S: 'static = (), T = mz_repr::Timestamp> {
 impl<T> DataflowDescription<Plan<T>, (), mz_repr::Timestamp> {
     /// Tests if the dataflow refers to a single timestamp, namely
     /// that `as_of` has a single coordinate and that the `until`
-    /// value corresponds to the `as_of` value plus one.
+    /// value corresponds to the `as_of` value plus one, or `as_of`
+    /// is the maximum timestamp and is thus single.
     pub fn is_single_time(&self) -> bool {
         // TODO: this would be much easier to check if `until` was a strict lower bound,
         // and we would be testing that `until == as_of`.
         let Some(as_of) = self.as_of.as_ref() else {
             return false;
         };
-        !as_of.is_empty()
-            && as_of
-                .as_option()
-                .and_then(|as_of| as_of.checked_add(1))
-                .as_ref()
-                == self.until.as_option()
+        let Some(as_of) = as_of.as_option() else {
+            return false;
+        };
+        if as_of == &mz_repr::Timestamp::MAX {
+            soft_assert_or_log!(
+                self.until.is_empty(),
+                "expected empty until due to MAX as_of, got {:?}",
+                self.until
+            );
+            return true;
```

Comment:
Is this fragment being added for readability purposes? If the `as_of` is `MAX`, then `try_step_forward` will return `None`. Also, only if `until` is empty will `as_option` on it return `None`. So this branch is already covered by the condition `as_of.try_step_forward().as_ref() == self.until.as_option()`, no?

Reply:
I suggested to add this, but maybe after adding `try_step_forward` this is no longer needed.